examples/vae/main.py

```diff
@@ -0,0 +1,147 @@
+from absl import app
+from absl import flags
+
+import jax.numpy as jnp
+import numpy as np
+
+import jax
+from jax import nn as jnn
+from jax import random
+
+from flax import nn
+from flax import optim
+
+import tensorflow_datasets as tfds
+
+from utils import save_image
+
+
+EPS = 1e-8
+KEY = random.PRNGKey(0)
```

Comment:
this is a bit of a subtle issue but creating jax arrays before the flags are parsed results in incorrect behavior when JAX/XLA depends on flags. So you need to create the PRNGKey in the main function

Reply:
I understand.
I removed this line and put `KEY = random.PRNGKey(0)` into the first line of `main()` function.